src/training/flava_eval/finetune_glue.py

```diff
@@ -211,6 +264,47 @@ def compute_metrics(model, dataloader, device, args):
     metrics["loss"] = val_loss / samples_seen
     return metrics
 
+def train_n_steps(model, data, optimizer, scheduler, early_stop, device, args):
+    model.train()
+    progress_bar = tqdm(total=args.num_steps)
+    data_train = iter(data['train'])
+    for i in range(args.num_steps):
+        scheduler(i)
+
+        try:
+            batch = next(data_train)
+        except StopIteration:
+            data = get_task_dataloaders(args)
+            data_train = iter(data['train'])
```

Comment:
Instead of getting the task_dataloaders again can you do the following with the dataloader: `data_iter = itertools.cycle(train_dataloader)` and use that to yield batches.